src/MessageList/MessageListBox.vala

```diff
@@ -97,30 +97,33 @@ public class Mail.MessageListBox : Gtk.ListBox {
         }
     }
 
-    public void add_inline_composer (ComposerWidget.Type type) {
+    public void add_inline_composer (ComposerWidget.Type type, MessageListItem? message_item = null) {
+        /* Can't open a new composer if thread is empty or currently has a composer open */
         var last_child = get_row_at_index ((int) get_children ().length () - 1);
-        var is_composer = (last_child != null && last_child is InlineComposer);
+        if (last_child == null || last_child is InlineComposer) {
+            return;
+        }
+
+        if (message_item == null) {
+            message_item = (MessageListItem) last_child;
+        }
+
         string content_to_quote = "";
         Camel.MimeMessage? mime_message = null;
         Camel.MessageInfo? message_info = null;
-        if (last_child is MessageListItem) {
```

Comment:
Not possible for `last_child` to be neither `InlineComposer` nor `MessageListItem`.